lib/src/main/java/edu/utexas/tacc/tapis/files/lib/caches/TenantAdminCache.java

```diff
@@ -0,0 +1,84 @@
+package edu.utexas.tacc.tapis.files.lib.caches;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import edu.utexas.tacc.tapis.files.lib.exceptions.ServiceException;
+import edu.utexas.tacc.tapis.files.lib.utils.LibUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+
+public class TenantAdminCache extends BasePermsCache {
+    private final LoadingCache<TenantAdminCache.TenantAdminCacheKey, Boolean> cache;
+    private static final Logger log = LoggerFactory.getLogger(FilePermsCache.class);
+
+    public TenantAdminCache() {
+        cache = CacheBuilder.newBuilder().expireAfterWrite(Duration.ofSeconds(60)).build(new TenantAdminLoader());
+    }
+
+    public Boolean checkPerm(String tenantId, String username)
+            throws ServiceException
+    {
+        try {
+            TenantAdminCacheKey key = new TenantAdminCacheKey(tenantId, username);
+            return cache.get(key);
+        } catch (ExecutionException ex) {
+            // TODO:  Wrong message - correct this
+            String msg = LibUtils.getMsg("FILES_CACHE_ERR", "FilePerms", tenantId, username, ex.getMessage());
```

Comment:
Just realized I still have a "TODO" to correct this message.  I'll take care of that.